state/sshconnrequests.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"strings"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/mgo/v3"
+	"github.com/juju/mgo/v3/txn"
+
+	"github.com/juju/juju/core/network"
+)
+
+type sshConnRequestDoc struct {
+	DocId               string    `bson:"_id"`
+	UnitName            string    `bson:"unit_name"`
+	Expires             time.Time `bson:"expires"`
+	Username            string    `bson:"username"`
+	Password            string    `bson:"password"`
+	ControllerAddresses []address `bson:"address"`
+	UnitPort            int       `bson:"unit_port"`
+	EphemeralPublicKey  []byte    `bson:"controller_public_key"`
+}
+
+type SSHConnRequest struct {
+	*sshConnRequestDoc
+}
+
+// SSHConnRequestArg holds the necessary info to create a ssh connection requests.
+type SSHConnRequestArg struct {
+	TunnelID           string
+	ModelUUID          string
+	UnitName           string
+	Expires            time.Time
+	Username           string
+	Password           string
+	ControllerAddress  network.SpaceAddresses
+	UnitPort           int
+	EphemeralPublicKey []byte
+}
+
+// SSHConnRequestRemoveArg holds the necessary info to remove a ssh connection requests.
+type SSHConnRequestRemoveArg struct {
+	TunnelID  string
+	ModelUUID string
+	UnitName  string
+}
+
+func sshReqConnKeyID(unitName string, id string) string {
```

Comment:
can we rename `id` to `tunnelID`, I also thought it was the model UUID for a second.